enhancements/samples-generation.md

```diff
@@ -0,0 +1,613 @@
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+1. Should not the project has a Memcached GO Sample with webhooks and other without using webhooks since it shows an advanced feature and it used is not mandatory? Note the Go tutorial do not use webhooks.  
+
+Answer: It should, since the memcached project is a sample for users to reference and webhooks are commonly used.
+
+## Summary
+
+The [operator-sdk-samples](https://github.com/operator-framework/operator-sdk-samples) repo contains an example Go operator project for users to reference when building their own projects. These example becomes outdated quickly because it requires its own release process and manual updates.
+
+Instead, this example project should be made generate-able via a code and (re)generated in a directory of the operator-sdk repo. Each PR to the operator-sdk repo that changes the scaffolds will require these samples be (re)generated via a makefile target and the CI should ensure that the samples are not outdated.
+
+Maintainers will be able to develop the code to generate the samples via the Golang language and using the same approach adopted on the tests. The helpers and utilities to mock the data will be available to be used for both.
+
+The e2e test also will be able to invoke the samples code implementation which will ensure that they are runnable and will keep centralized this implementation as well to achieve a better maintainability.
+
+In this way, it is no longer required keep the repository [operator-sdk-samples](https://github.com/operator-framework/operator-sdk-samples) and it should be deprecated. 
+
+**NOTE** The test utils as what is proposed here can fit well in upstream. It is not a Goal push these implementations to upstream, however, it definitely could be an ultimate end goal to keep in mind. 
+
+## Motivation
+
+### Goals
+
+- Generate samples via Go to improve the maintainability, readability and reusability
+- Checking if samples are outdated via CI to reduce its risk of the sample projects gets outdated
+- Building samples on SDK to facilitate the development process to mock data and keep the samples maintained by using the same approach and design defined and used via the test
+- Re-using samples on tests to improve the maintainability and ensure that they are runnable.
+- Helpers and utilities available for tests and samples
+- Specific implementation following the same approach to define the sample and the tests context
+
+### Non-Goals
+
+- Link the samples in the docs or aligned then with the tutorials
+- Provide advanced helpers or Boilerplates to mock the data that could be used in the tests and/or in the samples generation
+- Push this implementation to upstream
+
+## Proposal
+
+### User Stories
+
+#### Generate samples via Go 
+
+- I am a maintainer/contributor, I want to be able to troubleshoot the code used to create the samples, so that I can easily fix the issues. 
```

Comment:
If we are going to scaffold the operator from scratch and test all the functionalities, then would we be removing the existing e2e tests for go/ansible/helm? We would be doing the same steps twice.

Reply:
Using the samples for the e2e tests is an idea on the table, but when we talked about this initially we thought that the e2e tests should be out of scope for this proposal to keep it from becoming a huge task.